bundles/io/org.eclipse.smarthome.io.transport.mqtt/src/main/java/org/eclipse/smarthome/io/transport/mqtt/MqttService.java

```diff
@@ -228,28 +295,31 @@ public boolean addBrokerConnection(MqttBrokerConnection connection) {
     }
 
     /**
-     * Add a broker by a configuration key-value map. You need to provide at least a "name" and an "url".
+     * Add a broker by a configuration.
+     *
+     * <p>
+     * You need to provide at least a "name" and an "url".
      * Additional properties are "user","pwd","qos","retain","lwt","keepAlive","clientId", please read the
```

Comment:
Maybe it would be useful to mention the new properties here ( lwtTopic, lwtMessage lwtQos lwtRetain)

Reply:
The implementation has been changed to use a config object that properties are "fixed". As all fields are nullable and so optional the documentation needs to state which ones are required.
I don't know if it make sense to sync the additional properties all the time.
Perhaps we could remove the "Additional properties ..." part at all if this PR will be accepted (and before it is merged).